server/src/test/java/io/crate/metadata/cluster/AlterTableClusterStateExecutorTest.java

```diff
@@ -102,4 +105,42 @@ public void testAddExistingMetaDoNotAddEmptyValues() throws IOException {
         // DO NOT WANT empty containers and nulls: "{"_meta":{"indices":{},"partitioned_by":[],"primary_keys":null,"check_constraints":null,"constraints":{}}}"
     }
 
+    @Test
+    public void testAddExistingMetaCarriesOverRoutingColumnFromCurrentMeta() throws IOException {
+        Map<String, Object> mapping = new HashMap<>();
+        AlterTableRequest request = new AlterTableRequest(RelationName.fromIndexName("dummy"),
+                                                          null,
+                                                          true,
+                                                          true,
+                                                          Settings.EMPTY,
+                                                          Collections.singletonMap("_meta", mapping));
+
+        var currentMeta = new HashMap<String,Object>();
+        assertThat(AlterTableClusterStateExecutor.addExistingMeta(request, currentMeta),
+                   containsString("{\"_meta\":{}}"));
+
+        currentMeta.put("routing", "routing_col");
+        assertThat(AlterTableClusterStateExecutor.addExistingMeta(request, currentMeta),
+                   containsString("{\"_meta\":{\"routing\":\"routing_col\"}}"));
+    }
+
+    @Test
+    public void testAddExistingMetaDoesNotAllowToModifyRoutingColumn() throws IOException{
+        Map<String, Object> mapping = new HashMap<>();
+        mapping.put("routing", "routing_col_update");
+        var currentMeta = new HashMap<String,Object>();
+        AlterTableRequest request = new AlterTableRequest(RelationName.fromIndexName("dummy"),
+                                                           null,
+                                                           true,
+                                                           true,
+                                                           Settings.EMPTY,
+                                                           Collections.singletonMap("_meta", mapping));
+
+        Asserts.assertThrowsMatches(
+            () -> AlterTableClusterStateExecutor.addExistingMeta(request, currentMeta),
+            AssertionError.class,
+            "should not be possible to change routing_column"
+        );
+    }
```

Comment:
If we have a test case for this scenario I think we should replace the `assert` with a `if ... throw IllegalArgumentException`.

`asserts` can be disabled - although we generally have them enabled for tests it would be good to not rely on them for tests to pass.

Reply:
Thanks, exception seems more feasible here.